### PR TITLE
Adds typescript support and vue/typescript linting #10516

### PR DESCRIPTION
**Pull request by chrabyrd**
### Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Issues Solved

#10516

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+module.exports = {
```
Comment (jacobtylerwalls) [inline .eslintrc.js:1]: thanks for getting this wired up! 🎉 

Noticed spinning up on `arches-rdm` that moving the dotfiles up a directory causes `updateproject` to throw like this:

```py
  File "/Users/jwalls/prj/arches/arches/management/commands/updateproject.py", line 35, in update_to_v7
    shutil.copy2(os.path.join(settings.ROOT_DIR, dotfile), settings.APP_ROOT)
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/shutil.py", line 463, in copy2
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/shutil.py", line 260, in copyfile
    with open(src, 'rb') as fsrc:
         ^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/Users/jwalls/prj/arches/arches/.eslintrc.js'```

Comment (chrabyrd) [reply .eslintrc.js:1]: Yes, this is still in draft mode and I'm making my final round of checks before putting it out for review

**Review by apeters — COMMENTED** [review]
I'm getting this error when using "this" in templates
```Unexpected usage of 'this'              vue/this-in-template```

**Review by apeters — CHANGES_REQUESTED** [review]

File: arches/install/arches-templates/project_name/src/declarations.d.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// import declarations from other projects or Arches core
+import('../node_modules/arches/arches/app/src/declarations.d.ts');
```
[the remark this answers is not in the document]
Comment (chrabyrd) [reply arches/install/arches-templates/project_name/src/declarations.d.ts:2]: fixed in latest

File: package.json [hunk 3]
```diff
@@ -1,13 +1,18 @@
...
-        "build_production": "./arches/app/media/node_modules/.bin/cross-env NODE_PATH=./arches/app/media/node_modules NODE_OPTIONS=--max-old-space-size=2048 NODE_ENV=production ./arches/app/media/node_modules/.bin/webpack --config arches/webpack/webpack.config.prod.js",
-        "build_development": "./arches/app/media/node_modules/.bin/cross-env NODE_PATH=./arches/app/media/node_modules NODE_OPTIONS=--max-old-space-size=2048 ./arches/app/media/node_modules/.bin/webpack --config arches/webpack/webpack.config.dev.js",
-        "build_test": "./arches/app/media/node_modules/.bin/cross-env NODE_PATH=./arches/app/media/node_modules NODE_OPTIONS=--max-old-space-size=2048 ./arches/app/media/node_modules/.bin/webpack --config arches/webpack/webpack.config.dev.js --env test=true",
-        "start": "./arches/app/media/node_modules/.bin/cross-env NODE_PATH=./arches/app/media/node_modules NODE_OPTIONS=--max-old-space-size=2048 ./arches/app/media/node_modules/.bin/webpack serve --config arches/webpack/webpack.config.dev.js"
+        "typescript:check": "./arches/app/media/node_modules/.bin/vue-tsc --noEmit",
```
[the remark this answers is not in the document]
Comment (chrabyrd) [reply package.json]: fixed in latest

**Comment by chrabyrd** [comment]
> I'm getting this error when using "this" in templates `Unexpected usage of 'this' vue/this-in-template`

fixed in lateset

**Pull request merged**